lib/web/userpreferences.go

```diff
@@ -64,15 +64,24 @@ type AccessGraphPreferencesResponse struct {
 	HasBeenRedirected bool `json:"hasBeenRedirected"`
 }
 
+type DiscoverGuidePreferences struct {
+	PinnedGuides []string `json:"pinnedGuides"`
+}
+
+type DiscoverResourcePreferencesResponse struct {
+	DiscoverGuidePreferences *DiscoverGuidePreferences `json:"discoverGuidePreferences"`
+}
+
 // UserPreferencesResponse is the JSON response for the user preferences.
 type UserPreferencesResponse struct {
-	Assist                     AssistUserPreferencesResponse       `json:"assist"`
-	Theme                      userpreferencesv1.Theme             `json:"theme"`
-	UnifiedResourcePreferences UnifiedResourcePreferencesResponse  `json:"unifiedResourcePreferences"`
-	Onboard                    OnboardUserPreferencesResponse      `json:"onboard"`
-	ClusterPreferences         ClusterUserPreferencesResponse      `json:"clusterPreferences,omitempty"`
-	AccessGraph                AccessGraphPreferencesResponse      `json:"accessGraph,omitempty"`
-	SideNavDrawerMode          userpreferencesv1.SideNavDrawerMode `json:"sideNavDrawerMode"`
+	Assist                      AssistUserPreferencesResponse       `json:"assist"`
+	Theme                       userpreferencesv1.Theme             `json:"theme"`
+	UnifiedResourcePreferences  UnifiedResourcePreferencesResponse  `json:"unifiedResourcePreferences"`
+	Onboard                     OnboardUserPreferencesResponse      `json:"onboard"`
+	ClusterPreferences          ClusterUserPreferencesResponse      `json:"clusterPreferences,omitempty"`
+	DiscoverResourcePreferences DiscoverResourcePreferencesResponse `json:"discoverResourcePreferences"`
+	AccessGraph                 AccessGraphPreferencesResponse      `json:"accessGraph,omitempty"`
```

Comment:
should `DiscoverResourcePreferences` also define `omitempty` or we are ok sending empty fields?

Reply:
recording slack discussion: DiscoverResourcePreferences should always be initialized since i am using the `undefined` value as a way to determine if response came from an older proxy